core/src/lib.rs

```diff
@@ -341,6 +381,76 @@ mod test {
         assert!(attr3.flags() & AttributeBitFlags::IndexFulltext as u8 != 0);
         assert!(attr3.flags() & AttributeBitFlags::UniqueValue as u8 != 0);
     }
+
+    #[test]
+    fn test_as_edn_value() {
+        let mut schema = Schema::default();
+
+        let attr1 = Attribute {
+            index: true,
+            value_type: ValueType::Ref,
+            fulltext: false,
+            unique: None,
+            multival: false,
+            component: false,
+        };
+        associate_ident(&mut schema, NamespacedKeyword::new("foo", "bar"), 97);
+        add_attribute(&mut schema, 97, attr1);
+
+        let attr2 = Attribute {
+            index: false,
+            value_type: ValueType::String,
+            fulltext: true,
+            unique: Some(attribute::Unique::Value),
+            multival: true,
+            component: false,
+        };
+        associate_ident(&mut schema, NamespacedKeyword::new("foo", "bas"), 98);
+        add_attribute(&mut schema, 98, attr2);
+
+        let attr3 = Attribute {
+            index: false,
+            value_type: ValueType::Boolean,
+            fulltext: false,
+            unique: Some(attribute::Unique::Identity),
+            multival: false,
+            component: true,
+        };
+
+        associate_ident(&mut schema, NamespacedKeyword::new("foo", "bat"), 99);
+        add_attribute(&mut schema, 99, attr3);
+
+        let value = schema.as_edn_value();
+
+        let expected_output = r#"[ {   :db/id     :97
+    :db/ident     :foo/bar
+    :db/valueType :db.type/ref
+    :db/cardinality :db.cardinality/one
+    :db/index true },
+{   :db/id     :98
+    :db/ident     :foo/bas
+    :db/valueType :db.type/string
+    :db/cardinality :db.cardinality/many
+    :db/unique :db.unique/identity
+    :db/fulltext true },
+{   :db/id     :99
+    :db/ident     :foo/bat
+    :db/valueType :db.type/boolean
+    :db/cardinality :db.cardinality/one
+    :db/component true }, ]"#;
+        let expected_value = edn::parse::value(&expected_output).unwrap().without_spans();
+        assert_eq!(expected_value, value);
+    }
+
+    fn associate_ident(schema: &mut Schema, i: NamespacedKeyword, e: Entid) {
```

Comment:
nit: put these before (all) the tests, just 'cuz the test set will grow downwards.